pdks/Gf180/gf180/pdk.py

```diff
@@ -0,0 +1,560 @@
+    "6.0V_NMOS": _cap_module("cap_nmos_06v0", 3, GF180CapParams),
+    "6.0V_PMOS": _cap_module("cap_pmos_06v0", 3, GF180CapParams),
+    "3.3V_NMOS_Nwell": _cap_module("cap_nmos_03v3_b", 3, GF180CapParams),
+    "3.3V_PMOS_Pwell": _cap_module("cap_pmos_03v3_b", 3, GF180CapParams),
+    "6.0V_NMOS_Nwell": _cap_module("cap_nmos_06v0_b", 3, GF180CapParams),
+    "6.0V_PMOS_Pwell": _cap_module("cap_pmos_06v0_b", 3, GF180CapParams),
+}
+
+# Collected `ExternalModule`s are stored in the `modules` namespace
+modules = SimpleNamespace()
+# Add each to the `modules` namespace
+for mod in xtors.values():
+    setattr(modules, mod.name, mod)
+for mod in ress.values():
+    setattr(modules, mod.name, mod)
+for mod in caps.values():
+    setattr(modules, mod.name, mod)
+for mod in diodes.values():
+    setattr(modules, mod.name, mod)
+for mod in bjts.values():
+    setattr(modules, mod.name, mod)
+
+
+@dataclass
+class Cache:
+    """# Module-Scope Cache(s)"""
+
+    mos_modcalls: Dict[MosParams, h.ExternalModuleCall] = field(default_factory=dict)
+
+    res_modcalls: Dict[PhysicalResistorParams, h.ExternalModuleCall] = field(
+        default_factory=dict
+    )
+
+    cap_modcalls: Dict[PhysicalCapacitorParams, h.ExternalModuleCall] = field(
+        default_factory=dict
+    )
+
+    diode_modcalls: Dict[DiodeParams, h.ExternalModule] = field(default_factory=dict)
+
+    bjt_modcalls: Dict[BipolarParams, h.ExternalModule] = field(default_factory=dict)
+
+
+CACHE = Cache()
+
+
+class Gf180Walker(h.HierarchyWalker):
+    """Hierarchical Walker, converting `h.Primitive` instances to process-defined `ExternalModule`s."""
+
+    def __init__(self):
+        super().__init__()
+        # Caches of our external module calls
+        self.mos_modcalls = dict()
+        self.res_modcalls = dict()
+        self.cap_modcalls = dict()
+        self.diode_modcalls = dict()
+        self.bjt_modcalls = dict()
+
+    def visit_primitive_call(self, call: h.PrimitiveCall) -> h.Instantiable:
+        # Replace transistors
+        if call.prim is Mos:
+            return self.mos_module_call(call.params)
+
+        elif call.prim is PhysicalResistor or call.prim is ThreeTerminalResistor:
+            return self.res_module_call(call.params)
+
+        elif call.prim is PhysicalCapacitor or call.prim is ThreeTerminalCapacitor:
+            return self.cap_module_call(call.params)
+
+        elif call.prim is Diode:
+            return self.diode_module_call(call.params)
+
+        elif call.prim is Bipolar or call.prim is FourTerminalBipolar:
+            return self.bjt_module_call(call.params)
+
+        else:
+            raise RuntimeError(f"{call.prim} is not legitimate primitive")
+
+        return call
+
+    def mos_module(self, params: MosParams) -> h.ExternalModule:
+        """Retrieve or create an `ExternalModule` for a MOS of parameters `params`."""
+        mod = xtors.get((params.model, params.tp), None)
```

Comment:
Similar comment to the Sky130 PR here - it seems the implication is that every generic primitive _must_ set a `model` field to be compiled into this tech. 

While I haven't used this technology, there should, in general, be some notion of "the regular transistors" - i.e. the ones the standard logic library uses. The ideal of having different thresholds `vth` also generally transfers between techs. We should reserve the `model` field as an escape hatch for "weirder", more tech-specific things - e.g. the difference between having 6V and 20V devices, when the "main" ones are 1.8V.